lxqt-session/src/lxqtmodman.cpp

```diff
@@ -319,7 +319,25 @@ void LXQtModuleManager::restartModules(int exitCode, QProcess::ExitStatus exitSt
 LXQtModuleManager::~LXQtModuleManager()
 {
     qApp->removeNativeEventFilter(this);
-    qDeleteAll(mNameMap);
+
+    // We disconnect the finished signal before deleting the process. We do
+    // this to prevent a crash that results from a state change signal being
+    // emmited while deleting a crashing module.
+    // If the module is still connect restartModules will be called with a
+    // invalid sender.
+
+    ModulesMapIterator i(mNameMap);
+    while (i.hasNext())
+    {
+        i.next();
+
+        auto p = i.value();
+        disconnect(p, SIGNAL(finished(int, QProcess::ExitStatus)), 0, 0);
+
+        delete p;
+        p = 0;
```

Comment:
`p` is not a reference to the pointer, so the assignement is noop here (`p` goes out of scope immediately).